lib/dedupe.js

```diff
@@ -1,19 +1,18 @@
 // dedupe duplicated packages, or find them in the tree
-const util = require('util')
 const Arborist = require('@npmcli/arborist')
-const rimraf = util.promisify(require('rimraf'))
-const reifyOutput = require('./utils/reify-output.js')
 const usageUtil = require('./utils/usage.js')
 
 const usage = usageUtil('dedupe', 'npm dedupe')
-
 const completion = (cb) => cb(null, [])
 
-const cmd = (args, cb) => dedupe(args).then(() => cb()).catch(cb)
+const cmd = (args, cb) => dedupe().then(() => cb()).catch(cb)
 
-const dedupe = async args => {
-  require('npmlog').warn('coming soon!')
-  throw new Error('not yet implemented')
+const dedupe = async () => {
+  const arb = new Arborist({
+    ...npm.flatOptions,
+    path: where
+  })
+  await arb.dedupe()
```

Comment:
We will need to call `reifyOutput(arb)` after the dedupe is done.

Also, right now you can run dedupe in dry-run mode by running `npm find-dupes`.  Dry-run output isn't yet handled by the reifyOutput method, but it will need to be prior to official release.  In the meantime, I think we can do something like `new Arborist({ ...npm.flatOptions, dryRun: npm.flatOptions.dryRun || npm.command === 'find-dupes', path: where })`.